infrastructure/persistence/inventory/inventory_repository.go

```diff
@@ -89,8 +89,24 @@ func (c *InventoryClient) GetInventoryClients(ctx context.Context, page int) (In
 	}
 	defer res.Body.Close()
 
-	err = json.NewDecoder(res.Body).Decode(&results)
-	if err != nil {
+	if res.StatusCode != http.StatusOK {
+		responseBody, err := io.ReadAll(res.Body)
+		if err != nil {
+			log.Error().Err(err).Msg("Error reading response body")
+			return results, err
+		}
+
+		var errMsg error
+		switch {
+		case res.StatusCode >= 400:
+			errMsg = fmt.Errorf("error response received: %d, response body: %s", res.StatusCode, string(responseBody))
+			return results, errMsg
+		case res.StatusCode >= 300:
+			return results, nil
+		}
+	}
```

Comment:
I still think this code can be cleaned up to more clearly indicate its purpose. The status code checking can be organized so that the "happy path" flows through while still allowing the switch statement to handle future error statuses differently. It might appear pedantic now since all errors are handled the same, but in the future, you might want to handle HTTP 500 errors differently from HTTP 401 or 404s.

Here is one such suggestion. I haven't tested this code, but I believe it should follow the flow I described.

```suggestion
        switch {
        case res.StatusCode >= 400:
            responseBody, err := io.ReadAll(res.Body)
            if err != nil {
                log.Error().Err(err).Msg("Error reading response body")
                return results, fmt.Errorf("Error reading response body: %w", err)
            }
            defer responseBody.Close()
            
            return results, fmt.Errorf("error response received: %v, response body: %v", res.StatusCode, string(responseBody))
        }
```